pxr/imaging/hdMtlx/hdMtlx.cpp

```diff
@@ -358,7 +358,11 @@ _AddMaterialXNode(
         // Explicitly setting the node def string appparently fixes the problem.
         // If we don't do this code gen may fail.
         if (mxNode->getNodeDefString().empty()) {
-            mxNode->setNodeDefString(hdNodeType.GetText());
+            // Only set the NodeDef string if the node exists in the SdrRegistry.
+            // If it does not exist, this is a deleted MaterialX 1.38 node.
+            if (SdrRegistry::GetInstance().GetShaderNodeByIdentifier(hdNodeName)) {
+                mxNode->setNodeDefString(hdNodeType.GetText());
+            }
```

Comment:
Not needed if the MaterialX fix in https://github.com/AcademySoftwareFoundation/MaterialX/pull/2282 is used. Would require updating to MaterialX 1.39.4.